joda-money/src/main/java/com/fasterxml/jackson/datatype/jodamoney/MoneySerializer.java

```diff
@@ -50,11 +52,7 @@ private final void _writeFields(final Money money,
             final SerializerProvider context)
         throws IOException
     {
-        final BigDecimal decimal = money.getAmount();
-        final int decimalPlaces = money.getCurrencyUnit().getDecimalPlaces();
-        final int scale = Math.max(decimal.scale(), decimalPlaces);
-        g.writeNumberField("amount", decimal.setScale(scale, RoundingMode.UNNECESSARY));
-        g.writeFieldName("currency");
-        context.defaultSerializeValue(money.getCurrencyUnit(), g);
+        g.writeObjectField("amount", amountRepresenter.write(money));
```

Comment:
But doesn't this always chnge value from JSON Number into JSON String? And be essentially backwards-incompatible change?

This is why I think instead of `write()` conversion method it should be method that does write.
So, first `g.writeFieldName("amount")`, then value write by helper object.

Reply:
Ah. I did miss the paramaterization, yes, `Long` and `BigDecimal` will then be written using configured serializer which is fine.